In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("isl_player_final.csv")



In [3]:
df


tour_id tour_name      id  jersey_no                            name  \
0        148      ISL6    1514          3                    Asamoah Gyan   
1        148      ISL6    2475         26            Deshorn Dwayne Brown   
2        148      ISL6    2900          9                  Amine Chermiti   
3        148      ISL6    3753          7           Francisco Medina Luna   
4        148      ISL6    3809          4              Rafael Lopez Gomez   
..       ...       ...     ...        ...                             ...   
558      202      ISL7  144517          6  Hernan Daniel Santana Trujillo   
559      202      ISL7  144593          2                   Scott Neville   
560      202      ISL7  144594          4                      Daniel Fox   
561      202      ISL7  144804         30          Ningombam Engson Singh   
562      202      ISL7  145060         44              Haobam Tomba Singh   

             short_name  position_id    position position_short  team_id  ...  \
0          Asamoah Gyan            2     Forward            FWD      504  ...   
1         Deshorn Brown            2     Forward            FWD      656  ...   
2        Amine Chermiti            2     Forward            FWD      506  ...   
3                  Piti            3  Midfielder             MF     1159  ...   
4    Rafael Lopez Gomez            1    Defender            DEF     1536  ...   
..                  ...          ...         ...            ...      ...  ...   
558      Hernan Santana            3  Midfielder             MF      506  ...   
559       Scott Neville            1    Defender            DEF     1102  ...   
560          Daniel Fox            1    Defender            DEF     1102  ...   
561      N Engson Singh            3  Midfielder             MF     1874  ...   
562        Haobam Singh            3  Midfielder             MF     1102  ...   

    goaltenders.catches goaltenders.punches  \
0                     0                   0   
1                     0                   0   
2                     0                   0   
3                     0                   0   
4                     0                   0   
..                  ...                 ...   
558                   0                   0   
559                   0                   0   
560                   0                   0   
561                   0                   0   
562                   0                   0   

     goaltenders.penalty_kicks.shots_faced  \
0                                        0   
1                                        0   
2                                        0   
3                                        0   
4                                        0   
..                                     ...   
558                                      0   
559                                      0   
560                                      0   
561                                      0   
562                                      0   

     goaltenders.penalty_kicks.goals_allowed  goaltenders.penalty_kicks.saves  \
0                                          0                                0   
1                                          0                                0   
2                                          0                                0   
3                                          0                                0   
4                                          0                                0   
..                                       ...                              ...   
558                                        0                                0   
559                                        0                                0   
560                                        0                                0   
561                                        0                                0   
562                                        0                                0   

     country_id 

In [4]:
df.drop(["tour_id", "id", "name", "position_id", "position", "team_id", "team_short_name"], inplace = True, axis = 1)

In [5]:
pd.set_option("display.max_rows", None)

In [6]:
df

tour_name  jersey_no                short_name position_short  \
0        ISL6          3              Asamoah Gyan            FWD   
1        ISL6         26             Deshorn Brown            FWD   
2        ISL6          9            Amine Chermiti            FWD   
3        ISL6          7                      Piti             MF   
4        ISL6          4        Rafael Lopez Gomez            DEF   
5        ISL6         30           Anas Edathodika            DEF   
6        ISL6         17            Dhanpal Ganesh             MF   
7        ISL6         18            Semboi  Haokip            FWD   
8        ISL6         13          Lalrempuia Fanai             MF   
9        ISL6          1            Amrinder Singh             GK   
10       ISL6         24               Salam Singh            DEF   
11       ISL6          3             Albert Serran            DEF   
12       ISL6          5          Diawandou Diagne             MF   
13       ISL6         23            Carlos Delgado            DEF   
14       ISL6         14             Dimas Delgado             MF   
15       ISL6          5                    Juanan            DEF   
16       ISL6         22           Sergio Cidoncha             MF   
17       ISL6         12           Raphael Augusto             MF   
18       ISL6         28              Modou Sougou            FWD   
19       ISL6         27            Andre Schembri            FWD   
20       ISL6          9           Nerijus Valskis            FWD   
21       ISL6         17               Carlos Pena            DEF   
22       ISL6         13      Francisco Dorronsoro             GK   
23       ISL6         19        Halicharan Narzary             MF   
24       ISL6         33       Laxmikant Kattimani             GK   
25       ISL6         71         Mandar Rao Dessai             MF   
26       ISL6         21               Narayan Das            DEF   
27       ISL6         10             Paulo Machado             MF   
28       ISL6         33                Prabir Das            DEF   
29       ISL6         17             Pronay Halder             MF   
30       ISL6         19           Romeo Fernandes             MF   
31       ISL6         20             Mohammed Rafi             MF   
32       ISL6          8          Mohammed Rafique             MF   
33       ISL6         24             Subhasish Roy             GK   
34       ISL6          5                 Adil Khan             MF   
35       ISL6         23        Sauvik Chakrabarti            DEF   
36       ISL6         26             Shouvik Ghosh            DEF   
37       ISL6         40            Kunzang Bhutia             GK   
38       ISL6         13               TP Rehenesh             GK   
39       ISL6         23              Pritam Singh            DEF   
40       ISL6          8               Milan Singh             MF   
41       ISL6         16              Robin Gurung            DEF   
42       ISL6         22              Redeem Tlang             MF   
43       ISL6          7         Seiminlen Doungel             MF   
44       ISL6         10          Harmanjot Khabra            DEF   
45       ISL6         16               Jayesh Rane            FWD   
46       ISL6         24           Lenny Rodrigues             MF   
47       ISL6         20              Pritam Kotal            DEF   
48       ISL6         29      Arindam Bhattacharja             GK   
49       ISL6          1              Subrata Paul             GK   
50       ISL6          5              Raju Gaikwad            DEF   
51       ISL6          7          Mouhamadou Gning             MF   
52       ISL6          1            Karanjit Singh             GK   
53       ISL6          1     Gurpreet Singh Sandhu             GK   
54       ISL6         21               Roy Krishna            FWD   
55       ISL6          9             Sergio Castel            FWD   
56       ISL6         14      Nestor Jesus Benitez             MF   
57

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 563 entries, 0 to 562
Data columns (total 86 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   tour_name                                563 non-null    object 
 1   jersey_no                                563 non-null    int64  
 2   short_name                               563 non-null    object 
 3   position_short                           563 non-null    object 
 4   team_name                                563 non-null    object 
 5   is_started                               563 non-null    int64  
 6   is_onbench                               563 non-null    int64  
 7   is_goalkeeper                            563 non-null    int64  
 8   is_substitute                            563 non-null    int64  
 9   minutes_played                           563 non-null    int64  
 10  bonus_points                             563 non-n

In [8]:
cat_col = []
num_col = []

for i in df.columns:
    
    if df[i].dtypes == object:
        cat_col.append(i)
        
    else:
        num_col.append(i)
       
        

In [9]:
cat_col

['tour_name',
 'short_name',
 'position_short',
 'team_name',
 'country_name',
 'dob',
 'player_foot']

In [10]:
num_col

['jersey_no',
 'is_started',
 'is_onbench',
 'is_goalkeeper',
 'is_substitute',
 'minutes_played',
 'bonus_points',
 'injury_minutes_played',
 'actual_minutes_played',
 'goaltenders',
 'events.goals',
 'events.own_goals',
 'events.assists',
 'events.key_passes',
 'events.chances_created',
 'events.shots',
 'events.shots_on_target',
 'events.shots_off_target',
 'events.fouls_committed',
 'events.fouls_suffered',
 'events.yellow_cards',
 'events.red_cards',
 'events.is_second_yellow_card',
 'events.offsides',
 'events.corner_kicks',
 'events.crosses',
 'events.free_kicks',
 'events.throw_in',
 'events.punches',
 'events.catches',
 'events.goals_conceded',
 'events.clean_sheet',
 'events.cleansheet',
 'events.fantasy_assists',
 'events.penalties_saved',
 'events.penalties_earned',
 'events.penalties_conceded',
 'events.penalties_missed',
 'events.freekicks_conceded_goal',
 'events.freekicks_earned_goal',
 'events.blocked_shots',
 'events.defensive_blocked_shots',
 'events.penalty_kicks.to

In [11]:
df.columns

Index(['tour_name', 'jersey_no', 'short_name', 'position_short', 'team_name',
       'is_started', 'is_onbench', 'is_goalkeeper', 'is_substitute',
       'minutes_played', 'bonus_points', 'injury_minutes_played',
       'actual_minutes_played', 'goaltenders', 'events.goals',
       'events.own_goals', 'events.assists', 'events.key_passes',
       'events.chances_created', 'events.shots', 'events.shots_on_target',
       'events.shots_off_target', 'events.fouls_committed',
       'events.fouls_suffered', 'events.yellow_cards', 'events.red_cards',
       'events.is_second_yellow_card', 'events.offsides',
       'events.corner_kicks', 'events.crosses', 'events.free_kicks',
       'events.throw_in', 'events.punches', 'events.catches',
       'events.goals_conceded', 'events.clean_sheet', 'events.cleansheet',
       'events.fantasy_assists', 'events.penalties_saved',
       'events.penalties_earned', 'events.penalties_conceded',
       'events.penalties_missed', 'events.freekicks_conceded_g

In [12]:
df.groupby("team_name").agg({"events.goals" : ["mean", "median", "sum"]}).reset_index

<bound method DataFrame.reset_index of                     events.goals           
                            mean median sum
team_name                                  
ATK FC                  1.461538      0  38
ATK Mohun Bagan         1.214286      0  34
Bengaluru FC            1.016949      0  60
Chennaiyin FC           1.166667      0  56
FC Goa                  1.717391      0  79
Hyderabad FC            0.827586      0  48
Jamshedpur FC           0.750000      0  39
Kerala Blasters FC      0.894737      0  51
Mumbai City FC          1.254902      0  64
NorthEast United FC     0.732143      0  41
Odisha FC               0.877551      0  43
SC East Bengal          0.545455      0  18>

The team with the most goals is Hyderabad FC

In [13]:
grouped_cols = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"events.goals": ["min", "max", "sum"]})

In [14]:
pd.set_option("display.max_rows", None)

In [15]:
grouped_cols

events.goals  \
                                                                               min   
tour_name team_name           short_name               position_short                
ISL6      ATK FC              Agustin Iniguez          DEF                       0   
                              Anas Edathodika          DEF                       0   
                              Ankit Mukherjee          DEF                       0   
                              Arindam Bhattacharja     GK                        0   
                              Armando Sosa Pena        MF                        0   
                              Balwant Singh            FWD                       1   
                              Carl McHugh              MF                        0   
                              David Williams           FWD                       7   
                              Dheeraj Singh            GK                        0   
                              Edu Garcia               FWD                       6   
                              Javier Hernandez         MF                        2   
                              Jayesh Rane              FWD                       1   
                              Jobby Justin             FWD                       1   
                              John Johnson             DEF                       0   
                              Komal Thatal             FWD                       0   
                              Michael Soosairaj        FWD                       3   
                              Prabir Das               DEF                       0   
                              Pritam Kotal             DEF                       1   
                              Pronay Halder            MF                        1   
                              Regin Michael            MF                        0   
                              Ricky Lallawmawma        DEF                       0   
                              Roy Krishna              FWD                      15   
                              Salam Singh              DEF                       0   
                              Sehnaj Singh             MF                        0   
                              Sumit Rathi              DEF                       0   
                              Victor Mongil            DEF                       0   
          Bengaluru FC        Aditya Patra             GK                        0   
                              Albert Serran            DEF                       1   
                              Ashique Kuruniyan        FWD                       1   
                              Deshorn Brown            FWD                       3   
                              Dimas Delgado            MF                        1   
                              Edmund Lalrindika        FWD                       0   
                              Erik Paartalu            MF                        2   
                              Eugeneson Lyngdoh        MF                        0   
                              Francisco Borges         DEF                       0   
                              Gurpreet Singh Sandhu    GK                        0   
                              Gursimrat Singh Gill     DEF                       0   
                              Harmanjot Khabra         DEF                       0   
                              Juanan                   DEF                       1   
                              Kean Lewis               MF                        0   
                              Kevaughn  Frater         FWD                       1   
                              Leon Augustin            FWD                       0   
                              Manuel Onwu              FWD                       7   
                              Nishu Kumar              DEF                       1   
                            

The players from each team with the most goals in ISL6 are:
    
Roy Krishna from ATK FC with 15 goals - FWD
Nerijus Valskis from Chennaiyin FC with 15 goals - FWD
Bartholomew Ogbeche from Kerala Blasters with 15 goals - FWD

Ferran Corominas from FC Goa with 14 goals - FWD

Aridane Cabrera	from Odisha FC with 9 goals - FWD
Sunil Chhetri from Bengaluru with 9 goals - FWD

Marcelo Pereira from Hyderabad FC with 7 goals - FWD
Sergio Castel from Jamshedpur FC with 7 goals - FWD

Amine Chermiti from Mumbai City FC with 6 goals - FWD

Asamoah Gyan from NorthEast United FC with 4 goals - FWD




In [16]:
grouped_cols_forminutesplayed = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"actual_minutes_played": ["sum"]})

grouped_cols_forminutesplayed

 

actual_minutes_played
                                                                                        sum
tour_name team_name           short_name               position_short                      
ISL6      ATK FC              Agustin Iniguez          DEF                             1154
                              Anas Edathodika          DEF                              577
                              Ankit Mukherjee          DEF                                8
                              Arindam Bhattacharja     GK                              1918
                              Armando Sosa Pena        MF                               824
                              Balwant Singh            FWD                              157
                              Carl McHugh              MF                               442
                              David Williams           FWD                             1461
                              Dheeraj Singh            GK                                95
                              Edu Garcia               FWD                             1068
                              Javier Hernandez         MF                              1690
                              Jayesh Rane              FWD                              969
                              Jobby Justin             FWD                              357
                              John Johnson             DEF                              381
                              Komal Thatal             FWD                               12
                              Michael Soosairaj        FWD                             1599
                              Prabir Das               DEF                             1841
                              Pritam Kotal             DEF                             2013
                              Pronay Halder            MF                               171
                              Regin Michael            MF                               388
                              Ricky Lallawmawma        DEF                                0
                              Roy Krishna              FWD                             1938
                              Salam Singh              DEF                              356
                              Sehnaj Singh             MF                               419
                              Sumit Rathi              DEF                             1245
                              Victor Mongil            DEF                              667
          Bengaluru FC        Aditya Patra             GK                                 0
                              Albert Serran            DEF                             1450
                              Ashique Kuruniyan        FWD                             1338
                              Deshorn Brown            FWD                              385
                              Dimas Delgado            MF                              1865
                              Edmund Lalrindika        FWD                               11
                              Erik Paartalu            MF                              1556
                              Eugeneson Lyngdoh        MF                                98
                              Francisco Borges         DEF                              334
                              Gurpreet Singh Sandhu    GK                              1813
                              Gursimrat Singh Gill     DEF                              101
                              Harmanjot Khabra         DEF                             1685
                              Juanan                   DEF                             1809
                              Kean Lewis               MF                                52
                              Kevaughn  Frater         FWD                              202
                      

In [17]:
grouped_cols_forchancescreated = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"events.chances_created": ["sum"]})

grouped_cols_forchancescreated



events.chances_created
                                                                                         sum
tour_name team_name           short_name               position_short                       
ISL6      ATK FC              Agustin Iniguez          DEF                                 4
                              Anas Edathodika          DEF                                 0
                              Ankit Mukherjee          DEF                                 0
                              Arindam Bhattacharja     GK                                  0
                              Armando Sosa Pena        MF                                  9
                              Balwant Singh            FWD                                 0
                              Carl McHugh              MF                                  0
                              David Williams           FWD                                26
                              Dheeraj Singh            GK                                  0
                              Edu Garcia               FWD                                15
                              Javier Hernandez         MF                                 29
                              Jayesh Rane              FWD                                13
                              Jobby Justin             FWD                                 4
                              John Johnson             DEF                                 1
                              Komal Thatal             FWD                                 1
                              Michael Soosairaj        FWD                                 9
                              Prabir Das               DEF                                26
                              Pritam Kotal             DEF                                 4
                              Pronay Halder            MF                                  3
                              Regin Michael            MF                                  2
                              Ricky Lallawmawma        DEF                                 0
                              Roy Krishna              FWD                                31
                              Salam Singh              DEF                                 0
                              Sehnaj Singh             MF                                  2
                              Sumit Rathi              DEF                                 5
                              Victor Mongil            DEF                                 2
          Bengaluru FC        Aditya Patra             GK                                  0
                              Albert Serran            DEF                                 1
                              Ashique Kuruniyan        FWD                                11
                              Deshorn Brown            FWD                                 3
                              Dimas Delgado            MF                                 40
                              Edmund Lalrindika        FWD                                 0
                              Erik Paartalu            MF                                 18
                              Eugeneson Lyngdoh        MF                                  1
                              Francisco Borges         DEF                                 7
                              Gurpreet Singh Sandhu    GK                                  0
                              Gursimrat Singh Gill     DEF                                 0
                              Harmanjot Khabra         DEF                                 6
                              Juanan                   DEF                                 5
                              Kean Lewis               MF                                  1
                              Kevaughn  Frater         FWD             

In [18]:
grouped_cols_forshots = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"events.shots": ["sum"]})

grouped_cols_forshots


events.shots
                                                                               sum
tour_name team_name           short_name               position_short             
ISL6      ATK FC              Agustin Iniguez          DEF                       8
                              Anas Edathodika          DEF                       0
                              Ankit Mukherjee          DEF                       0
                              Arindam Bhattacharja     GK                        0
                              Armando Sosa Pena        MF                       10
                              Balwant Singh            FWD                       3
                              Carl McHugh              MF                        0
                              David Williams           FWD                      32
                              Dheeraj Singh            GK                        0
                              Edu Garcia               FWD                      34
                              Javier Hernandez         MF                       38
                              Jayesh Rane              FWD                      10
                              Jobby Justin             FWD                       5
                              John Johnson             DEF                       0
                              Komal Thatal             FWD                       2
                              Michael Soosairaj        FWD                      21
                              Prabir Das               DEF                       8
                              Pritam Kotal             DEF                       6
                              Pronay Halder            MF                        1
                              Regin Michael            MF                        1
                              Ricky Lallawmawma        DEF                       0
                              Roy Krishna              FWD                      53
                              Salam Singh              DEF                       1
                              Sehnaj Singh             MF                        4
                              Sumit Rathi              DEF                       2
                              Victor Mongil            DEF                       6
          Bengaluru FC        Aditya Patra             GK                        0
                              Albert Serran            DEF                       3
                              Ashique Kuruniyan        FWD                      23
                              Deshorn Brown            FWD                      19
                              Dimas Delgado            MF                       22
                              Edmund Lalrindika        FWD                       0
                              Erik Paartalu            MF                       18
                              Eugeneson Lyngdoh        MF                        0
                              Francisco Borges         DEF                      11
                              Gurpreet Singh Sandhu    GK                        0
                              Gursimrat Singh Gill     DEF                       0
                              Harmanjot Khabra         DEF                       4
                              Juanan                   DEF                      16
                              Kean Lewis               MF                        1
                              Kevaughn  Frater         FWD                       5
                              Leon Augustin            FWD                       2
                              Manuel Onwu              FWD                      14
                              Nishu Kumar              DEF                      14
                              Parag Srivas             DEF                       0
                              Prabhsukhan Singh Gill   GK                        0
   

In [19]:
grouped_cols_forshotsontarget = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"events.shots_on_target": ["sum"]})

grouped_cols_forshotsontarget


events.shots_on_target
                                                                                         sum
tour_name team_name           short_name               position_short                       
ISL6      ATK FC              Agustin Iniguez          DEF                                 2
                              Anas Edathodika          DEF                                 0
                              Ankit Mukherjee          DEF                                 0
                              Arindam Bhattacharja     GK                                  0
                              Armando Sosa Pena        MF                                  0
                              Balwant Singh            FWD                                 1
                              Carl McHugh              MF                                  0
                              David Williams           FWD                                19
                              Dheeraj Singh            GK                                  0
                              Edu Garcia               FWD                                13
                              Javier Hernandez         MF                                 12
                              Jayesh Rane              FWD                                 3
                              Jobby Justin             FWD                                 3
                              John Johnson             DEF                                 0
                              Komal Thatal             FWD                                 1
                              Michael Soosairaj        FWD                                 7
                              Prabir Das               DEF                                 2
                              Pritam Kotal             DEF                                 1
                              Pronay Halder            MF                                  1
                              Regin Michael            MF                                  0
                              Ricky Lallawmawma        DEF                                 0
                              Roy Krishna              FWD                                24
                              Salam Singh              DEF                                 0
                              Sehnaj Singh             MF                                  0
                              Sumit Rathi              DEF                                 1
                              Victor Mongil            DEF                                 1
          Bengaluru FC        Aditya Patra             GK                                  0
                              Albert Serran            DEF                                 1
                              Ashique Kuruniyan        FWD                                 9
                              Deshorn Brown            FWD                                 8
                              Dimas Delgado            MF                                  2
                              Edmund Lalrindika        FWD                                 0
                              Erik Paartalu            MF                                  8
                              Eugeneson Lyngdoh        MF                                  0
                              Francisco Borges         DEF                                 1
                              Gurpreet Singh Sandhu    GK                                  0
                              Gursimrat Singh Gill     DEF                                 0
                              Harmanjot Khabra         DEF                                 1
                              Juanan                   DEF                                 7
                              Kean Lewis               MF                                  0
                              Kevaughn  Frater         FWD             

In [20]:
grouped_cols_forassist = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"events.assists": ["sum"]})

grouped_cols_forassist


events.assists
                                                                                 sum
tour_name team_name           short_name               position_short               
ISL6      ATK FC              Agustin Iniguez          DEF                         0
                              Anas Edathodika          DEF                         0
                              Ankit Mukherjee          DEF                         0
                              Arindam Bhattacharja     GK                          0
                              Armando Sosa Pena        MF                          1
                              Balwant Singh            FWD                         0
                              Carl McHugh              MF                          0
                              David Williams           FWD                         5
                              Dheeraj Singh            GK                          0
                              Edu Garcia               FWD                         3
                              Javier Hernandez         MF                          5
                              Jayesh Rane              FWD                         3
                              Jobby Justin             FWD                         1
                              John Johnson             DEF                         0
                              Komal Thatal             FWD                         0
                              Michael Soosairaj        FWD                         0
                              Prabir Das               DEF                         5
                              Pritam Kotal             DEF                         0
                              Pronay Halder            MF                          1
                              Regin Michael            MF                          0
                              Ricky Lallawmawma        DEF                         0
                              Roy Krishna              FWD                         6
                              Salam Singh              DEF                         0
                              Sehnaj Singh             MF                          0
                              Sumit Rathi              DEF                         0
                              Victor Mongil            DEF                         0
          Bengaluru FC        Aditya Patra             GK                          0
                              Albert Serran            DEF                         0
                              Ashique Kuruniyan        FWD                         1
                              Deshorn Brown            FWD                         0
                              Dimas Delgado            MF                          5
                              Edmund Lalrindika        FWD                         0
                              Erik Paartalu            MF                          5
                              Eugeneson Lyngdoh        MF                          0
                              Francisco Borges         DEF                         1
                              Gurpreet Singh Sandhu    GK                          0
                              Gursimrat Singh Gill     DEF                         0
                              Harmanjot Khabra         DEF                         0
                              Juanan                   DEF                         0
                              Kean Lewis               MF                          0
                              Kevaughn  Frater         FWD                         0
                              Leon Augustin            FWD                         1
                              Manuel Onwu              FWD                         2
                              Nishu Kumar              DEF                         0
                              Parag Srivas             DEF                 

In [21]:
grouped_cols_forgoodpasses = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"touches.good_passes": ["sum"]})

grouped_cols_forgoodpasses


touches.good_passes
                                                                                      sum
tour_name team_name           short_name               position_short                    
ISL6      ATK FC              Agustin Iniguez          DEF                            214
                              Anas Edathodika          DEF                            102
                              Ankit Mukherjee          DEF                              0
                              Arindam Bhattacharja     GK                              70
                              Armando Sosa Pena        MF                             238
                              Balwant Singh            FWD                             16
                              Carl McHugh              MF                             113
                              David Williams           FWD                            253
                              Dheeraj Singh            GK                               4
                              Edu Garcia               FWD                            310
                              Javier Hernandez         MF                             431
                              Jayesh Rane              FWD                            230
                              Jobby Justin             FWD                             63
                              John Johnson             DEF                            105
                              Komal Thatal             FWD                              3
                              Michael Soosairaj        FWD                            358
                              Prabir Das               DEF                            291
                              Pritam Kotal             DEF                            503
                              Pronay Halder            MF                              49
                              Regin Michael            MF                             181
                              Ricky Lallawmawma        DEF                              0
                              Roy Krishna              FWD                            263
                              Salam Singh              DEF                             76
                              Sehnaj Singh             MF                              86
                              Sumit Rathi              DEF                            270
                              Victor Mongil            DEF                            147
          Bengaluru FC        Aditya Patra             GK                               0
                              Albert Serran            DEF                            361
                              Ashique Kuruniyan        FWD                            259
                              Deshorn Brown            FWD                             49
                              Dimas Delgado            MF                             903
                              Edmund Lalrindika        FWD                              2
                              Erik Paartalu            MF                             515
                              Eugeneson Lyngdoh        MF                              17
                              Francisco Borges         DEF                             69
                              Gurpreet Singh Sandhu    GK                             170
                              Gursimrat Singh Gill     DEF                             15
                              Harmanjot Khabra         DEF                            393
                              Juanan                   DEF                            394
                              Kean Lewis               MF                               6
                              Kevaughn  Frater         FWD                             28
                              Leon Augustin            FWD                             21
                    

In [22]:

df["shot_accuracy"] = df['events.shots_on_target']/df['events.shots'] * 100

df["assist_accuracy"] = df["events.assists"]/df["events.chances_created"] * 100

df["minplay_goalratio"] = df["actual_minutes_played"]/df["events.goals"]
df

tour_name  jersey_no                short_name position_short  \
0        ISL6          3              Asamoah Gyan            FWD   
1        ISL6         26             Deshorn Brown            FWD   
2        ISL6          9            Amine Chermiti            FWD   
3        ISL6          7                      Piti             MF   
4        ISL6          4        Rafael Lopez Gomez            DEF   
5        ISL6         30           Anas Edathodika            DEF   
6        ISL6         17            Dhanpal Ganesh             MF   
7        ISL6         18            Semboi  Haokip            FWD   
8        ISL6         13          Lalrempuia Fanai             MF   
9        ISL6          1            Amrinder Singh             GK   
10       ISL6         24               Salam Singh            DEF   
11       ISL6          3             Albert Serran            DEF   
12       ISL6          5          Diawandou Diagne             MF   
13       ISL6         23            Carlos Delgado            DEF   
14       ISL6         14             Dimas Delgado             MF   
15       ISL6          5                    Juanan            DEF   
16       ISL6         22           Sergio Cidoncha             MF   
17       ISL6         12           Raphael Augusto             MF   
18       ISL6         28              Modou Sougou            FWD   
19       ISL6         27            Andre Schembri            FWD   
20       ISL6          9           Nerijus Valskis            FWD   
21       ISL6         17               Carlos Pena            DEF   
22       ISL6         13      Francisco Dorronsoro             GK   
23       ISL6         19        Halicharan Narzary             MF   
24       ISL6         33       Laxmikant Kattimani             GK   
25       ISL6         71         Mandar Rao Dessai             MF   
26       ISL6         21               Narayan Das            DEF   
27       ISL6         10             Paulo Machado             MF   
28       ISL6         33                Prabir Das            DEF   
29       ISL6         17             Pronay Halder             MF   
30       ISL6         19           Romeo Fernandes             MF   
31       ISL6         20             Mohammed Rafi             MF   
32       ISL6          8          Mohammed Rafique             MF   
33       ISL6         24             Subhasish Roy             GK   
34       ISL6          5                 Adil Khan             MF   
35       ISL6         23        Sauvik Chakrabarti            DEF   
36       ISL6         26             Shouvik Ghosh            DEF   
37       ISL6         40            Kunzang Bhutia             GK   
38       ISL6         13               TP Rehenesh             GK   
39       ISL6         23              Pritam Singh            DEF   
40       ISL6          8               Milan Singh             MF   
41       ISL6         16              Robin Gurung            DEF   
42       ISL6         22              Redeem Tlang             MF   
43       ISL6          7         Seiminlen Doungel             MF   
44       ISL6         10          Harmanjot Khabra            DEF   
45       ISL6         16               Jayesh Rane            FWD   
46       ISL6         24           Lenny Rodrigues             MF   
47       ISL6         20              Pritam Kotal            DEF   
48       ISL6         29      Arindam Bhattacharja             GK   
49       ISL6          1              Subrata Paul             GK   
50       ISL6          5              Raju Gaikwad            DEF   
51       ISL6          7          Mouhamadou Gning             MF   
52       ISL6          1            Karanjit Singh             GK   
53       ISL6          1     Gurpreet Singh Sandhu             GK   
54       ISL6         21               Roy Krishna            FWD   
55       ISL6          9             Sergio Castel            FWD   
56       ISL6         14      Nestor Jesus Benitez             MF   
57

In [23]:
grouped_cols_shotacc = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"shot_accuracy": ["sum"]})

grouped_cols_shotacc


shot_accuracy
                                                                                sum
tour_name team_name           short_name               position_short              
ISL6      ATK FC              Agustin Iniguez          DEF                25.000000
                              Anas Edathodika          DEF                 0.000000
                              Ankit Mukherjee          DEF                 0.000000
                              Arindam Bhattacharja     GK                  0.000000
                              Armando Sosa Pena        MF                  0.000000
                              Balwant Singh            FWD                33.333333
                              Carl McHugh              MF                  0.000000
                              David Williams           FWD                59.375000
                              Dheeraj Singh            GK                  0.000000
                              Edu Garcia               FWD                38.235294
                              Javier Hernandez         MF                 31.578947
                              Jayesh Rane              FWD                30.000000
                              Jobby Justin             FWD                60.000000
                              John Johnson             DEF                 0.000000
                              Komal Thatal             FWD                50.000000
                              Michael Soosairaj        FWD                33.333333
                              Prabir Das               DEF                25.000000
                              Pritam Kotal             DEF                16.666667
                              Pronay Halder            MF                100.000000
                              Regin Michael            MF                  0.000000
                              Ricky Lallawmawma        DEF                 0.000000
                              Roy Krishna              FWD                45.283019
                              Salam Singh              DEF                 0.000000
                              Sehnaj Singh             MF                  0.000000
                              Sumit Rathi              DEF                50.000000
                              Victor Mongil            DEF                16.666667
          Bengaluru FC        Aditya Patra             GK                  0.000000
                              Albert Serran            DEF                33.333333
                              Ashique Kuruniyan        FWD                39.130435
                              Deshorn Brown            FWD                42.105263
                              Dimas Delgado            MF                  9.090909
                              Edmund Lalrindika        FWD                 0.000000
                              Erik Paartalu            MF                 44.444444
                              Eugeneson Lyngdoh        MF                  0.000000
                              Francisco Borges         DEF                 9.090909
                              Gurpreet Singh Sandhu    GK                  0.000000
                              Gursimrat Singh Gill     DEF                 0.000000
                              Harmanjot Khabra         DEF                25.000000
                              Juanan                   DEF                43.750000
                              Kean Lewis               MF                  0.000000
                              Kevaughn  Frater         FWD                80.000000
                              Leon Augustin            FWD                 0.000000
                              Manuel Onwu              FWD                85.714286
                              Nishu Kumar              DEF                42.857143
                              Parag Srivas             DEF                 0.000000
                              Prabhsuk

In [24]:
grouped_cols_assistacc = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"assist_accuracy": ["sum"]})

grouped_cols_assistacc


assist_accuracy
                                                                                  sum
tour_name team_name           short_name               position_short                
ISL6      ATK FC              Agustin Iniguez          DEF                   0.000000
                              Anas Edathodika          DEF                   0.000000
                              Ankit Mukherjee          DEF                   0.000000
                              Arindam Bhattacharja     GK                    0.000000
                              Armando Sosa Pena        MF                   11.111111
                              Balwant Singh            FWD                   0.000000
                              Carl McHugh              MF                    0.000000
                              David Williams           FWD                  19.230769
                              Dheeraj Singh            GK                    0.000000
                              Edu Garcia               FWD                  20.000000
                              Javier Hernandez         MF                   17.241379
                              Jayesh Rane              FWD                  23.076923
                              Jobby Justin             FWD                  25.000000
                              John Johnson             DEF                   0.000000
                              Komal Thatal             FWD                   0.000000
                              Michael Soosairaj        FWD                   0.000000
                              Prabir Das               DEF                  19.230769
                              Pritam Kotal             DEF                   0.000000
                              Pronay Halder            MF                   33.333333
                              Regin Michael            MF                    0.000000
                              Ricky Lallawmawma        DEF                   0.000000
                              Roy Krishna              FWD                  19.354839
                              Salam Singh              DEF                   0.000000
                              Sehnaj Singh             MF                    0.000000
                              Sumit Rathi              DEF                   0.000000
                              Victor Mongil            DEF                   0.000000
          Bengaluru FC        Aditya Patra             GK                    0.000000
                              Albert Serran            DEF                   0.000000
                              Ashique Kuruniyan        FWD                   9.090909
                              Deshorn Brown            FWD                   0.000000
                              Dimas Delgado            MF                   12.500000
                              Edmund Lalrindika        FWD                   0.000000
                              Erik Paartalu            MF                   27.777778
                              Eugeneson Lyngdoh        MF                    0.000000
                              Francisco Borges         DEF                  14.285714
                              Gurpreet Singh Sandhu    GK                    0.000000
                              Gursimrat Singh Gill     DEF                   0.000000
                              Harmanjot Khabra         DEF                   0.000000
                              Juanan                   DEF                   0.000000
                              Kean Lewis               MF                    0.000000
                              Kevaughn  Frater         FWD                   0.000000
                              Leon Augustin            FWD                  33.333333
                              Manuel Onwu              FWD                  25.000000
                              Nishu Kumar              DEF                   0.000000
                            

In [25]:
grouped_cols_ratio_of_minutesplayed_to_goal = df.groupby(["tour_name", "team_name", "short_name", "position_short"]).agg({"minplay_goalratio": ["sum"]})

grouped_cols_ratio_of_minutesplayed_to_goal

minplay_goalratio
                                                                                    sum
tour_name team_name           short_name               position_short                  
ISL6      ATK FC              Agustin Iniguez          DEF                          inf
                              Anas Edathodika          DEF                          inf
                              Ankit Mukherjee          DEF                          inf
                              Arindam Bhattacharja     GK                           inf
                              Armando Sosa Pena        MF                           inf
                              Balwant Singh            FWD                   157.000000
                              Carl McHugh              MF                           inf
                              David Williams           FWD                   208.714286
                              Dheeraj Singh            GK                           inf
                              Edu Garcia               FWD                   178.000000
                              Javier Hernandez         MF                    845.000000
                              Jayesh Rane              FWD                   969.000000
                              Jobby Justin             FWD                   357.000000
                              John Johnson             DEF                          inf
                              Komal Thatal             FWD                          inf
                              Michael Soosairaj        FWD                   533.000000
                              Prabir Das               DEF                          inf
                              Pritam Kotal             DEF                  2013.000000
                              Pronay Halder            MF                    171.000000
                              Regin Michael            MF                           inf
                              Ricky Lallawmawma        DEF                     0.000000
                              Roy Krishna              FWD                   129.200000
                              Salam Singh              DEF                          inf
                              Sehnaj Singh             MF                           inf
                              Sumit Rathi              DEF                          inf
                              Victor Mongil            DEF                          inf
          Bengaluru FC        Aditya Patra             GK                      0.000000
                              Albert Serran            DEF                  1450.000000
                              Ashique Kuruniyan        FWD                  1338.000000
                              Deshorn Brown            FWD                   128.333333
                              Dimas Delgado            MF                   1865.000000
                              Edmund Lalrindika        FWD                          inf
                              Erik Paartalu            MF                    778.000000
                              Eugeneson Lyngdoh        MF                           inf
                              Francisco Borges         DEF                          inf
                              Gurpreet Singh Sandhu    GK                           inf
                              Gursimrat Singh Gill     DEF                          inf
                              Harmanjot Khabra         DEF                          inf
                              Juanan                   DEF                  1809.000000
                              Kean Lewis               MF                           inf
                              Kevaughn  Frater         FWD                   202.000000
                              Leon Augustin            FWD                          inf
                              Manuel Onwu              FWD                   107.714286
                      

The top 3 contenders for the best forward of the Indian Super League season 6 are:

Roy Krishna from ATK FC 
Nerijus Valskis from Chennaiyin FC
Bartholomew Ogbeche from Kerala Blasters 



ROY KRISHNA                             NERIJUS VALSKIS                         BARTHOLOMEW OGBECHE     

Goals: 15                               Goals: 15                               Goals: 15
Minutes Played: 1938                    Minutes Played: 1727                    Minutes Played: 1411
    

Shots: 53                               Shots: 51                               Shots: 55
Shots on Target: 24                     Shots on Target: 24                     Shots on Target: 25

Chances Created: 31                     Chances Created: 23                     Chances Created: 9
Assist: 6                               Assist: 6                               Assist: 1

Shot Accuracy: 45.28%                   Shot Accuracy: 47.05%                   Shot Accuracy: 45.45%
Assist Accuracy: 19.35%                 Assist Accuracy: 26.09%                 Assist Accuracy: 11.11%
Minutes played to Goal Ratio: 129.00    Minutes played to Goal Ratio: 115.13    Minutes played to Goal Ratio: 94.06



IN THE ISL6
BEST FORWARD OF THE SEASON: Bartholomew Ogbeche



